### PR TITLE
itemcharges: track ardy cloak charges left

### LINKED ISSUE #14723: Ardy cloak/explorer ring teleport charges
can there be a counter of teleport charges on these items? i have no way of knowing how many charges are on them unless i actually teleport or like, manually keep track. if this feature already exists in game can someone show me how to see that?

### PR DESCRIPTION
**Pull request by SirGirion**
Partially closes #14723 w.r.t ardy cloak charges

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java [hunk 1]
```diff
@@ -547,6 +577,36 @@ public void onWidgetLoaded(WidgetLoaded widgetLoaded)
...
 	}
 
+	@Subscribe
+	private void onGameTick(GameTick event)
+	{
+		long currentTime = System.currentTimeMillis();
+		boolean dailyReset = !loggingIn && currentTime - lastReset > ONE_DAY;
+
+		if ((dailyReset || loggingIn)
+			&& client.getVar(VarClientInt.MEMBERSHIP_STATUS) == 1)
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java:587]: `Client#getVar()` is deprecated, use `Client#getVarcIntValue()` instead.
```suggestion
			&& client.getVarcIntValue(VarClientInt.MEMBERSHIP_STATUS) == 1)
```

**Review by Nightfirecat — COMMENTED** [review]
I can confirm the test strings are ok. This LGTM generally besides the one `getVar()` use.

**Review by Nightfirecat — COMMENTED** [review]
Needs rebase, but LGTM

File: runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java (hunk 1 above)
Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java:588]: This doesn't look resolved to me, perhaps you incorrectly resolved this branch's merge conflict. Please be sure you have resolved a review comment before resolving the conversation in GitHub's UI.

**Review by Nightfirecat — CHANGES_REQUESTED** [review]

File: runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java (hunk 1 above)
Comment (SirGirion) [reply runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java]: This should be correct now

**Comment by Nightfirecat** [comment]
I'm reverting this change as the daily reset portion of this code does not work as expected. (it does not reset charges unless the client has been open for a day or longer) I'll be working on this a bit more before re-committing it.

**Pull request merged**